app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt

```diff
@@ -13,25 +13,28 @@ import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.example.randomusergenerator.R
+import com.example.randomusergenerator.data.local.UserData
 import com.example.randomusergenerator.data.local.UserViewState
 import com.example.randomusergenerator.ui.resources.Dimensions.space_x2
 import com.example.randomusergenerator.user.viewmodel.UserViewModel
 
 @Composable
 fun UserScreen() {
-    val userViewModel: UserViewModel = viewModel()
+    val userViewModel: UserViewModel = hiltViewModel()
     val userViewState by userViewModel.uiState.collectAsState()
 
-    LaunchedEffect(key1 = true) {
+    LaunchedEffect(key1 = false) {
```

Comment:
Why change to `false` if I may ask? 🤔 
And why aren't you using a string as a key identifier?

Reply:
I was checking for why it always calls the getUser function whenever the user comes to first screen. Still same behaviour when the navigateUP is used to navigate back

Reply:
I will move this discussion to slack.

Reply:
okay